### PR TITLE
QArchive: Add version 2.2.2 and modernize

### PR DESCRIPTION
**Pull request by ericriff**
Add support for the latest version and modernize recipe and test packages.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by ericriff** [comment]
The dev kindly created a new tag with https://github.com/antony-jr/QArchive/pull/62 so I don't have to include that as a patch here

**Review by prince-chrismc — APPROVED** [review]

**Comment by antony-jr** [comment]
@ericriff v2.2.3 adds a commit for using **c++17 standard** when Qt6 is used, like how quazip does. Qt6 requires c++17 but we use c++11 by default in versions older than 2.2.3, so if you use v2.2.3 for conan, the package can be used with Qt6 also.

**Comment by ericriff** [comment]
I updated the recipe to use the new tagged version and I'm passing `QUAZIP_QT_MAJOR_VERSION` based on the version of Qt. Regardless, I don't like this pattern. I'm only following to be consistent with `Quazip` but I believe this should  be an option instead, something like `with_qt` that defaults to `5` but also `6` can be selected. Then we can change the `requires` accordingly (and also this CMake var)

**Comment by antony-jr** [comment]
> I updated the recipe to use the new tagged version and I'm passing `QARCHIVE_QT_VERSION_MAJOR` based on the version of Qt. Regardless, I don't like this pattern. I'm only following it to be consistent with `Quazip` but I believe this should  be an option instead, something like `with_qt` that defaults to `5` but also `6` can be selected. Then we can change the `requires` accordingly (and also this CMake var). As it is now, if you want `Qarchive/Quazip` with `qt6` you have to edit the recipe.

In QArchive build, If the cmake variable is not set then cmake uses the default version of Qt installed in the system, the variable is only needed to be set if both qt5 and qt6 is installed, to select between the two.

**Review by prince-chrismc — DISMISSED** [review]

**Comment by prince-chrismc** [comment]
What's wrong with the consumer overriding the qt dep to what ever version they want?

I mean `--requires` and `--option` will do the same thing and result in different pkg ids 🤔 and the UX is the same "I pick this version of Qt" 🤷 this is less likely to conflict with other recipes which support different majors

**Comment by ericriff** [comment]
I was thinking about having to edit the actual conanfile, but you're right, you can override it without editing the recipe. Please disregard my comment.

File: recipes/qarchive/all/conanfile.py [hunk 1]
```diff
@@ -29,14 +29,11 @@ class QarchiveConan(ConanFile):
         "fPIC": True,
     }
 
-    generators = "cmake", "cmake_find_package"
-
     @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def _qt_major(self):
+        return Version(self.deps_cpp_info["qt"].version).major
```
Comment (uilianries) [inline recipes/qarchive/all/conanfile.py:34]:
```suggestion
        return Version(self.dependencies["qt"].ref.version).major
```

https://docs.conan.io/en/latest/migrating_to_2.0/recipes.html#the-generate-method
https://docs.conan.io/en/latest/reference/conanfile/dependencies.html#dependencies-interface

**Review by uilianries — COMMENTED** [review]
Only a small change to keep aligned with the migration guide.

**Comment by ericriff** [comment]
Is this stuck? :thinking:

**Review by prince-chrismc — APPROVED** [review]

**Review by uilianries — APPROVED** [review]
LGTM

**Pull request merged**